.github/workflows/optimized-ci.yml

```
@@ -0,0 +1,379 @@
+    runs-on: ubuntu-latest
+    needs: [changes]
+    if: needs.changes.outputs.backend == 'true' || needs.changes.outputs.tests == 'true'
+    strategy:
+      fail-fast: false
+      matrix:
+        ci_node_index: [0, 1]
+        ci_node_total: [2]
+    
+    services:
+      postgres:
+        image: postgres:${{ env.POSTGRES_VERSION }}-alpine
```

actionlint (1.7.7)

context "env" is not allowed here. available contexts are "github", "inputs", "matrix", "needs", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)